app/localoperator/DieterichRuinaAgeing.h

```diff
@@ -86,37 +86,54 @@ class DieterichRuinaAgeing {
         double snAbs = -sn + p_[index].get<SnPre>();
         double tauAbs = norm(tauAbsVec);
         double V = 0.0;
-        if (eta == 0.0) {
+        double max_xi = -std::pow(2, 8);
+        double a, b;
+        if (eta == 0.0)
+        {
             V = Finv(index, snAbs, tauAbs, psi);
-        } else {
-            if (snAbs <= 0.0) { /* Implies the fault is experiencing tension / opening */
+        }
+        else
+        {
+            if (snAbs <= 0.0)
+            { /* Implies the fault is experiencing tension / opening */
                 snAbs = 0.0; /* Just to illustrate what we are doing */
                 /* Solve R(V) = T - sigma_n F(V,psi) - eta V with sigma_n = 0.0 */
                 V = tauAbs / eta;
-            } else {
-                double a = 0.0;
-                double b = tauAbs / eta;
-                if (a > b) {
-                    std::swap(a, b);
-                }
-                auto fF = [this, &index, &snAbs, &tauAbs, &psi, &eta](double V) {
-                    return tauAbs - this->F(index, snAbs, V, psi) - eta * V;
+            }
+            else
+            {
+                b = std::log10(tauAbs / eta);
+                auto fF = [this, &index, &snAbs, &tauAbs, &psi, &eta](double Ve)
+                {
+                    return tauAbs - this->F(index, snAbs, std::pow(10.0,Ve), psi) - eta * std::pow(10.0,Ve);
                 };
-                try {
-                    V = zeroIn(a, b, fF);
-                } catch (std::exception const&) {
-                    std::cout << "sigma_n = " << snAbs << std::endl
-                              << "|tau| = " << tauAbs << std::endl
-                              << "psi = " << psi << std::endl
-                              << "L = " << a << std::endl
-                              << "U = " << b << std::endl
-                              << "F(L) = " << fF(a) << std::endl
-                              << "F(U) = " << fF(b) << std::endl;
-                    throw;
-                }
+                for (int xi = 5; xi < 9; ++xi)
+                {
+                    try
+                    {
+                        a = -std::pow(2.0, xi);
+                        auto Ve = zeroIn(a, b, fF);
+                        V = std::pow(10.0, Ve);
+                        break;
+                    }
+                    catch (std::exception const&)
+                    {
+                        if (a == max_xi)
```

Comment:
Equality comparisons between doubles / floats are almost never going to be true. Don't rely on them. I would use (if you adopt changes above)

```
if (xi == max_xi) {
```

Reply:
done